src/pages/test-report/components/reverse-diagram-modal/reverse-diagram-modal.ts

```diff
@@ -143,4 +152,33 @@ export class ReverseDiagramCatDPage implements OnInit {
     this.store$.dispatch(new ReverseDiagramWidthChanged(this.vehicleWidth, vehicleWidth));
     this.calculateReversingWidth(vehicleWidth);
   }
+
+  getReducerInfo(testCategory: TestCategory) {
+    switch (testCategory){
+      case TestCategory.C:
+      case TestCategory.C1:
+      case TestCategory.C1E:
+      case TestCategory.CE:
+        return {
+          vehicleDetails: getVehicleDetailsC,
+          vehicleWidth: getVehicleWidthC,
+          vehicleLength: getVehicleLengthC,
+        };
+      case TestCategory.D1:
+      case TestCategory.D1E:
+      case TestCategory.D:
+      case TestCategory.DE:
+        return {
+          vehicleDetails: getVehicleDetailsD,
+          vehicleWidth: getVehicleWidthD,
+          vehicleLength: getVehicleLengthD,
+        };
+      default:
+        return {
+          vehicleDetails: getVehicleDetailsBE,
+          vehicleWidth: getVehicleWidthBE,
+          vehicleLength: getVehicleLengthBE,
+        };
+    }
+  }
```

Comment:
Can we extract this into a provider as we have done for the test-data-by-category

Reply:
Sure, will add it to the Acceptance Criteria for https://jira.dvsacloud.uk/browse/MES-4854

Reply:
AC for above story edited